projects/addon-mobile/directives/dropdown-mobile/dropdown-mobile.directive.ts

```diff
@@ -16,6 +18,13 @@ import {TuiDropdownMobileComponent} from './dropdown-mobile.component';
                     ? TuiDropdownMobileComponent
                     : inject(TUI_DROPDOWN_COMPONENT, {skipSelf: true}),
         },
+        {
+            provide: TUI_OPTION_CONTENT,
+            useFactory: () =>
+                inject(TUI_IS_MOBILE)
+                    ? TuiOptionMobile
```

Comment:
Inject also `TuiTextfieldComponent` for backward compatibility for legacy controls